### PR TITLE
Fixes #13211: Create a notification plugin

### PR DESCRIPTION
**Pull request by VictorQrt**
https://www.rudder-project.org/redmine/issues/13211

### CODE REVIEW

**Comment by VictorQrt** [comment]
Commit modified

File: rudder-notify/bin/notify.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            self.notify_slack(msg)
+
+    def notify_slack(self, msg):
+        for webhook in self.conf["SLACK"]["webhooks"].split(' '):
+            requests.post(webhook, headers={"Content-type":"application/json"}, data=json.dumps({
+                "text":"*RUDDER NON-COMPLIANCE*\n" + str(msg)
+                }))
+
+    def notify_mail(self):
+        if self.conf["MAIL"]["nospam"] == "true" and time.time() - self.timestamp < self.get_mail_batch_period():
```
Comment (peckpeck) [inline rudder-notify/bin/notify.py]: I'd like a comment here to say this is the batching code and what it does

File: rudder-notify/bin/notify.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    # This function reads its input from the Rudder server non-compliance logs directly.
+    # It is a temporary solution, used until the server's non-compliance hooks are implemented.
+    def run_with_logtail(self):
+        for line in log_tail.log_tail("/var/log/rudder/compliance/non-compliant-reports.log"):
+            self.handle_non_compliance(line)
+
+    def start(self):
+        self.run()
+        #self.run_with_logtail()
```
Comment (peckpeck) [inline rudder-notify/bin/notify.py]: The opposite

File: rudder-notify/bin/notify.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            return
+        
+        tmp_mailfile = "/tmp/rudder-notify-mail.txt"
+        with open(tmp_mailfile, 'a') as out:
+            out.write(str(len(self.notif_queue)) + " notification" + ('s' if len(self.notif_queue) > 1 else '') + " from Rudder :\n")
+            for i in range(0, len(self.notif_queue)):
+                out.write("# " + str(i+1) + ' :\n' + str(self.notif_queue[i]) + '\n')
+        self.notif_queue = []
+        for recipient in self.conf["MAIL"]["recipients"].split(' '):
+            os.system("cat " + tmp_mailfile + " | mail -s 'Rudder non-compliance notification' " + recipient)
```
Comment (peckpeck) [inline rudder-notify/bin/notify.py]: cat | ?

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

**Pull request merged**